### DISCUSSION TITLE
Importing data into existing tables

### DISCUSSION THREAD

**Discussion by kgodey** [mathesar-foundation/mathesar#2269] [Feature Ideas]
> [!TIP]
> **Edit:** This feature is planned to be included in an upcoming Mathesar release. There is no timeline yet, but progress can be tracked on the project issue: #4618 

Users can import CSV files into Mathesar to create a new table. We’d like to extend this functionality so that they can import data into an existing table. We already have much of the backend work needed for this feature.

---
*If you'd like us to build this feature, please comment below with more details about how you plan to use this functionality.*

**Comment by MrChadMWood** [comment]
This feature would be very valuable for importing large amounts of rows very quickly.
I think right now, rows must be entered one at a time via the GUI.

May I suggest allowing users to manually map the table column names to the csv column headers during import.

**Comment by mathemancer** [comment]
FWIW, the back end for this is done, and includes the column mapping functionality requested by @MrChadMWood . We also suggest a mapping based on column names and types.

**Comment by jeromeperez** [comment]
Hello.
This feature would be very interesting for my users because they do 2 campaigns by year and add a lot of data each time.
But sometimes they modify existing data in the same campaign.

**Reply by kgodey** [reply]
Thank you for letting us know @jeromeperez. it will help us prioritize this feature sooner.

**Reply by jeromeperez** [reply]
this feature is very important for my user, i hope it would be implemented soon ;-)

**Comment by kha84** [comment]
Looks like this is getting some traction recently - https://github.com/mathesar-foundation/mathesar/issues/4618  Hurray! 👍
